### PR TITLE
logging: add config load/setup/docs

### PR DESCRIPTION
**Pull request by katcipis**
Add basic config loading + docs on how to use logging. here is an example playing around with it:

```
#!/bin/bash

set -o errexit
set -o nounset

echo "=== default ==="
echo

terramate version


echo
echo "=== debug enabled, console/coloring by default ==="
echo

export TM_LOG=DEBUG

terramate version

echo
echo "=== debug enabled, disabling coloring ==="
echo

export TM_LOG_COLOR=OFF

terramate version

echo
echo "=== JSON logging ==="
echo

export TM_LOG_FMT=JSON

terramate version
```

There is a minimal amount of debug for now on the main entrypoint, useful to test this, then we can add more.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,42 @@ on how different kinds of configurations can help you to avoid duplication
 and write solid Infrastructure as Code.
 
 
+## Debugging
+
+Terramate has detailed logs which can be enabled by setting the **TM\_LOG**
+environment variable to any of the following values:
```
Comment (mariux) [inline README.md:94]: please use cmd line option on top. 

```suggestion
Terramate has detailed logs which can be enabled by setting the `TM_LOG_LEVEL`
environment variable to any of the following values:
```

File: README.md [hunk 2]
```diff
@@ -88,6 +88,42 @@ on how different kinds of configurations can help you to avoid duplication
...
+
+* DEBUG
+* INFO
+* WARN
+* ERROR
+* DISABLED
```
Comment (mariux) [inline README.md:100]:
```suggestion
* `DEBUG`
* `INFO`
* `WARN`
* `ERROR`
* `NONE`
```

File: README.md [hunk 3]
```diff
@@ -88,6 +88,42 @@ on how different kinds of configurations can help you to avoid duplication
...
+
+This will cause detailed logs to appear on stderr.
+
+To control the format of the logs you can use **TM\_LOG\_FMT**,
+setting it to any of the following values:
+
+* CONSOLE
+* JSON
```
Comment (mariux) [inline README.md:108]:
```suggestion
* `TEXT`
* `JSON`
```

File: README.md [hunk 4]
```diff
@@ -88,6 +88,42 @@ on how different kinds of configurations can help you to avoid duplication
...
+
+* DEBUG
+* INFO
+* WARN
+* ERROR
+* DISABLED
+
+This will cause detailed logs to appear on stderr.
+
+To control the format of the logs you can use **TM\_LOG\_FMT**,
```
Comment (mariux) [inline README.md:104]:
```suggestion
To control the format of the logs you can use `TM_LOG_FMT`,
```

**Review by mariux — COMMENTED** [review]
- please fix markdown fmt 
- and support commandline options over environment.

  `terramate --log-level=debug --log-fmt=json --no-color` is the more user friendly interface and matches existing tooling or even `--verbose/-v` and `--debug` or `--log-json` as we only have text/json

- color should default to no if running not in a tty...

**Review by i4ki — APPROVED** [review]
LGTM

**Comment by katcipis** [comment]
I got ahead of myself with this PR. The idea is for @kassianh to work on this (maybe some of this can be used, hope it helps :-) ).

**Pull request closed without merging**